### ISSUE TITLE
[Script]:  Graph Neural Network(GNNs) for Node Classification

### ISSUE THREAD

**Issue by smty2018** [avinashkranjan/Amazing-Python-Scripts#2910]
### Aim

a Graph Neural Network (GNN) for node classification on graph-structured data. The example uses the PyTorch Geometric library to build and train a GNN on the Cora dataset.

### Details

Dataset: The code uses the Cora dataset, which is a well-known citation network dataset. It consists of scientific publications as nodes in a graph, where edges represent citation relationships between publications. Each node (publication) has a bag-of-words feature representation and a class label indicating its research field.

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I'm a GSSoC'23 contributor
- [X] I want to work on this issue
